include/server.h

```diff
@@ -55,6 +55,11 @@ namespace kiwix
         */
        std::string getHttpUrl();
 
+       /**
+        * Get all available network interfaces
+        */
+       std::map<std::string, std::string> getNetworkInterfaces();
```

Comment:
I actually have no idea whats the correct way to do that.
Any hints/help please!

Reply:
Just remove it :) (The method is already exposed (in your PR) in `networkTools.h`)
 
If we really want to have it a `Server` level we must define the method as static (https://en.cppreference.com/w/cpp/language/static)

Reply:
But if it isn't available at a Server level (after removing), how would I use it if needed (like in https://github.com/kiwix/kiwix-desktop/pull/735)

Reply:
So, is this approach fine? 
Remove the includes (`networkTools.h`) and functions from internalServer
include `networkTools.h` in `server.cpp`
define `getNetworkInterfaces()` and `getBestPublicIp` as static  (?)

Reply:
> But if it isn't available at a Server level (after removing), how would I use it if needed

We simply keep the "plain" function in `networkTools.h`.
Then in kiwix-desktop, we simply have to include `networkTools.h` and use the function :
```cpp
#include <kiwix/networkTools.h>
[...]
const auto interfacesMap = kiwix::getNetworkInterfaces();
```

We CAN also define `getNetworkInterfaces` as a static method but it doesn't provide any useful value  (except than explicitly telling that the methode is associated to the "global" server feature).
But we MUST not define `getNetworkInterfaces` as a classic method. Simply because we want the user to be able to use this function before creating a server.

The simpler is to just declare it as a plain function in `networkTools.h` and remove it from the server class.

For `getBestPublicIp` it is different. Is is the best public IP to contact the server. So it must be associated to a server instance. And the returned value may change depending of how the server has been started.

Reply:
Thanks for this amazing explaination! 

One more query, for networkTool.h to be available via `#include <kiwix/networkTools.h`, it should be moved to `include/` directory, right?

Reply:
No, my bad.

The public functions must be declared in `include/tools.h`. `networkTools.h` (and friends) are for private helpers. You can take example on the functions already declared in `tools.h` and implemented in `<foo>Tools.cpp`.
And kiwix-desktop will `#include <kiwix/tools.h>`